app/types/scheduler.py

```diff
@@ -0,0 +1,229 @@
+    return await job_function(**kwargs)
+
+
+class Scheduler:
+    """
+    An [arq](https://arq-docs.helpmanual.io/) scheduler.
+    The wrapper is intended to be used inside a FastAPI worker.
+
+    The scheduler use a Redis database to store planned jobs.
+    """
+
+    # See https://github.com/fastapi/fastapi/discussions/9143#discussioncomment-5157572
+
+    def __init__(self):
+        # ArqWorker, in charge of scheduling and executing tasks
+        self.worker: Worker | None = None
+        # Task will contain the asyncio task that runs the worker
+        self.task: asyncio.Task | None = None
+        # Pointer to the get_db dependency
+        # self._get_db: Callable[[], AsyncGenerator[AsyncSession, None]]
+
+    async def start(
+        self,
+        redis_host: str,
+        redis_port: int,
+        redis_password: str | None,
+        _get_db: Callable[[], AsyncGenerator[AsyncSession, None]],
+        **kwargs,
+    ):
+        """
+        Parameters:
+        - redis_host: str
+        - redis_port: int
+        - redis_password: str
+        - get_db: Callable[[], AsyncGenerator[AsyncSession, None]] a pointer to `get_db` dependency
+        """
+
+        class ArqWorkerSettings(WorkerSettingsBase):
+            functions = [run_task]
+            allow_abort_jobs = True
+            keep_result_forever = True
+            redis_settings = RedisSettings(
+                host=redis_host,
+                port=redis_port,
+                password=redis_password or "",
+            )
+
+        # We pass handle_signals=False to avoid arq from handling signals
+        # See https://github.com/python-arq/arq/issues/182
+        self.worker = create_worker(
+            ArqWorkerSettings,
+            handle_signals=False,
+            **kwargs,
+        )
+        # We run the worker in an asyncio task
+        self.task = asyncio.create_task(self.worker.async_run())
+
+        self._get_db = _get_db
+
+        scheduler_logger.info("Scheduler started")
+
+    async def close(self):
+        # If the worker was started, we close it
+        if self.worker is not None:
+            await self.worker.close()
+
+    async def queue_job_time_defer(
+        self,
+        job_function: Callable[..., Coroutine[Any, Any, Any]],
+        job_id: str,
+        defer_seconds: int,
+        **kwargs,
+    ):
+        """
+        Queue a job to execute job_function in defer_seconds amount of seconds
+        job_id will allow to abort if needed
+        """
+        if self.worker is None:
+            scheduler_logger.error("Scheduler not started")
+            return None
+        job = await self.worker.pool.enqueue_job(
+            "run_task",
+            job_function=job_function,
+            _job_id=job_id,
+            _defer_by=timedelta(seconds=defer_seconds),
+            _get_db=self._get_db,
+            **kwargs,
+        )
+        scheduler_logger.debug(f"Job {job_id} queued {job}")
+
+    async def queue_job_defer_to(
```

Comment:
Is there a real interest in keeping both `queue_job_time_defer` and `queue_job_defer_to`? 
I think we are likely to calculate `defer_seconds` using a `timedelta` and `datetime`, and thus could easily use `queue_job_defer_to` instead.

It seems Arq will calculate a `datetime` in both cases, so there shouldn't be a performance gain from using a method or the other.

I would think only having one method will reduce code complexity and limit potentials mistakes